bokehjs/src/coffee/models/widgets/data_table.coffee

```diff
@@ -46,7 +47,7 @@ export class DataProvider
     return null
 
   getField: (offset, field) ->
-    if field == DTINDEX_NAME
+    if @model.index_column and (field == DTINDEX_NAME)
```

Comment:
not sure this is necessary -- if you have not added an index column then field should never be `DTINDEX_NAME` so I think the check is unnecessary